### PR TITLE
feat: Cognite3DModel.changeStyledNodeSetAppearance; feat: ByTreeIndexNodeSet - allow empty constructor

### PR DESCRIPTION
**Pull request by maksnester**

### CODE REVIEW

**Review by larsmoa — CHANGES_REQUESTED** [review]
Looks good - but can you also add a section before Examples in cad-styling.mdx that explains how you add/change/remove styled sets? I'm thinking an example where you e.g. style nodes with treeIndex 0-100.000 in a color, change the color of the styled set and finally remove it (?)

File: viewer/src/datamodels/cad/styling/NodeAppearanceProvider.test.ts [hunk 1]
```diff
@@ -61,16 +61,19 @@ describe('NodeAppearanceProvider', () => {
     expect(applyCb).toBeCalledWith(expect.anything(), expect.anything(), set1.getIndexSet(), style1);
   });
 
-  test('add and remove style triggers changed', () => {
+  test('add/change/remove style triggers called', () => {
```
[the remark this answers is not in the document]
Comment (maksnester) [reply viewer/src/datamodels/cad/styling/NodeAppearanceProvider.test.ts:64]: sure, good point 👍

File: documentation/docs/examples/cad-styling.mdx [hunk 2]
```diff
@@ -43,6 +43,28 @@ identifies 3D nodes by `treeIndex`. There are several implementations of `NodeSe
 It's also possible [implement custom `NodeSet`](cad-styling-custom). This can be useful for specialized queries or to load
 sets from custom data sources.
 
+This is how you can bind a specific appearance to your node set and update it:
```
Comment (larsmoa) [inline documentation/docs/examples/cad-styling.mdx:46]:
```suggestion
`Cognite3DModel.addStyledNodeSet`, `changeStyledNodeSetAppearance` and `removeStyledNodeSet` is used to manage styled set associated with a model:
```

**Review by larsmoa — APPROVED** [review]
See suggestions

**Pull request merged**